cli/azd/pkg/config/config.go

```diff
@@ -56,6 +64,45 @@ func (c *config) Raw() map[string]any {
 	return c.data
 }
 
+// SetSecret stores the secrets at the specified path within a local user vault
+func (c *config) SetSecret(path string, value string) error {
+	if c.vaultId == "" {
+		c.vault = NewConfig(nil)
+		c.vaultId = uuid.New().String()
+		if err := c.Set("vault", c.vaultId); err != nil {
+			return fmt.Errorf("failed setting vault id: %w", err)
+		}
+	}
+
+	pathId := uuid.New().String()
+	vaultRef := fmt.Sprintf("vault://%s/%s", c.vaultId, pathId)
+	if err := c.vault.Set(pathId, base64.StdEncoding.EncodeToString([]byte(value))); err != nil {
+		return fmt.Errorf("failed setting secret value: %w", err)
+	}
+
+	return c.Set(path, vaultRef)
```

Comment:
Wondering if we also want to update the implementation of `Raw()` such that it fetches the values from the vault, so you see actual values for the secret keys instead of vault references.

Reply:
Yes, we can do that if we detect it is a secrets... Happy to iterate on this if we want to take this style of implementation.